### DISCUSSION TITLE
Webp Frustration

### DISCUSSION THREAD

**Discussion by psyclonemedia** [ImageMagick/ImageMagick#6654] [Help]
### ImageMagick version

7.1

### Operating system, version and so on

Oracle Linux Server 7.9 (Based on RHEL 7)

### Description

We have had imagemagick for years.. Just went to upgrader from 6.9 to 7.1.x  with support for webp.  There is no issue in converting to webp from png or jpeg syste is doing that fine but if we try to upload a webp directly, imagemagick is not identifying the webp images and skipping these images entirely.

We are running a RoR app that doe some image management for a CMS, and we have installed libwebp.

any help would be greatly Appreciated

**Comment by snibgo** [comment]
> ... but if we try to upload a webp directly, imagemagick is not identifying the webp images and skipping these images entirely.

That is a vague description of a problem. Please show the command you used, and an input file that created the problem, and any error messages output.

Please also show the text output from "`magick -version`".

**Reply by mr-dev2** [reply]
magick -version

Version: ImageMagick 7.1.1-15 Q16-HDRI x86_64 21298 https://imagemagick.org
Copyright: (C) 1999 ImageMagick Studio LLC
License: https://imagemagick.org/script/license.php
Features: Cipher DPC HDRI OpenMP(3.1) 
Delegates (built-in): bzlib cairo fontconfig freetype jng jpeg lcms lzma openexr pangocairo png rsvg tiff x xml zlib
Compiler: gcc (4.8)

we installed libwebp using this command(sudo yum install libwebp). we had this version => libwebp-0.3.0-11.el7.x86_64

**Comment by snibgo** [comment]
`Delegates (built-in)` does not include webp. Check with `magick -list format`. This should include a line like this:
```
WEBP* rw+   WebP Image Format (libwebp 1.3.1 [020F])
```
I expect that shows your build of IM cannot read or write WEBP format.

You say you have libwebp-0.3.0. This seems old. I suggest you upgrade that, and re-build IM.

**Reply by psyclonemedia** [reply]
Thank you

**Reply by mr-dev2** [reply]
i got this when i try to upgrade it using this command (sudo yum -y update libwebp). Loaded plugins: langpacks, ulninfo
Package libwebp-0.3.0-11.el7.x86_64 already installed and latest version
Nothing to do.

please suggest how to install specific version if libwebp.

**Reply by mr-dev2** [reply]
i could see only these with w* with this command =>  magick -list format
 WBMP* rw-   Wireless Bitmap (level 0) image
 WEBM  rw+   Open Web Media
 WMV  rw+   Windows Media Video
 WPG* rw-   Word Perfect Graphics

**Comment by snibgo** [comment]
Do you build IM with `./config`? If you do, then `config.log` may give a clue. For example, for me it contains:
```
configure:34012: checking for libwebp >= 0.4.1
configure:34019: $PKG_CONFIG --exists --print-errors "libwebp >= 0.4.1"
configure:34022: $? = 0
configure:34036: $PKG_CONFIG --exists --print-errors "libwebp >= 0.4.1"
configure:34039: $? = 0
configure:34077: result: yes
```

I don't know how to install a more recent libwebp on your system.
